common/src/main/java/net/neoforged/gradle/common/extensions/JarJarExtension.java

```diff
@@ -35,9 +39,30 @@ public void enable() {
     }
 
     private void enable(boolean enabled) {
-        final Task task = project.getTasks().findByPath("jarJar");
+        if (this.enabled == enabled) {
+            return;
+        }
+        this.enabled = enabled;
+        final JarJar task = (JarJar) project.getTasks().findByPath("jarJar");
+        Configuration runtimeElements = project.getConfigurations().maybeCreate(JavaPlugin.RUNTIME_ELEMENTS_CONFIGURATION_NAME);
         if (task != null) {
-            task.setEnabled(enabled);
+            if (enabled) {
+                removedFromPublication.clear();
```

Comment:
Leaving a note to myself here; I have to actually add the new stuff to this list to make enabling jarJar reversible